### PR TITLE
Rework Getting Started/projects page

### PR DESCRIPTION
**Pull request by tomaskallup**
I had to change `projects.md` to `projects.mdx` and update astro (I did so via the recommended `npx @astrojs/upgrade`), which changed some type files inside `.astro`, to allow usage of their custom `<FileTree>` component, which renders a nice clickable file tree.

Other then that, I tried to make `projects.mdx` more of a "copy" of the Hello world, to make sure it includes information about building & running.

### CODE REVIEW

File: src/content/docs/Getting Started/projects.mdx [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+import { FileTree } from '@astrojs/starlight/components';
+
+:::note[Not installed the C3 compiler yet?]
+[Download C3](/install-c3/prebuilt-binaries/), available on Mac, Windows and Linux.
+:::
+
+## Projects in C3
+Projects are *optional*, but are a good way to manage compiling code when there
+ are a lot of files and modules.
+They also allow you so specify libraries to link and define how your project should be built for specific targets.
```
Comment (joshring) [inline src/content/docs/Getting Started/projects.mdx]:
```suggestion
They also allow you to specify libraries to link, and define how your project should be built for specific targets.
```

File: src/content/docs/Getting Started/projects.mdx [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+Let's go over some of the useful ones;
+- `./build` is where your temporary files and build results will end up
+- `./docs` are the overall documentations for your code
+- `./lib` is where C3 libraries (with the .c3l suffix) end up
+- `./resources` are for any resources like images, sound effects etc.
+- `./scripts` is for various scripts, including .c3 scripts that can run at compile time to generate code
+- `./src` is what we care about as it's where a majority of our code will go and it now contains `main.c3` with "Hello World" example
+- `project.json` file is where you add more data about your project, sort of like package.json for node.js
+- `LICENSE` file for your code license
+- `README.md` file where you can describe your project
```
Comment (joshring) [inline src/content/docs/Getting Started/projects.mdx]:
```suggestion
#### Directory Overview

| Directory       | Usage         |
|-----------------|---------------|
| `./build`       | Where your temporary files and build results will go. |
| `./docs`        | Code Documentation |
| `./lib`         | C3 libraries (with the .c3l suffix) |
| `./resources`   | Non-code esources like images, sound effects etc. |
| `./scripts`     | Scripts, including .c3 scripts that generate code at compile time. |
| `./src`         | Storing our code, by default contains `main.c3` with "Hello World". |
| `project.json`  | Record project information, similar to `package.json` in NodeJS. |
| `LICENSE`       | Project license. |
| `README.md`     | Help others understand and use your code. | 
```

File: src/content/docs/Getting Started/projects.mdx [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+```
+
+This command builds the project targets defined in our `project.json` file.
+
+If you want to build a specific target, you can do so by specifying its name.
+The default target is created with the name of the project, such as `myc3project`.
+
+```bash 
+c3c build myc3project
+```
```
Comment (joshring) [inline src/content/docs/Getting Started/projects.mdx:95]:
```suggestion
:::Note
If you want to build a specific target, you can do so by specifying its name.
The default target is created with the name of the project, such as `myc3project`.

```bash 
c3c build myc3project
```
:::
```

File: src/content/docs/Getting Started/projects.mdx [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+- `./src` is what we care about as it's where a majority of our code will go and it now contains `main.c3` with "Hello World" example
+- `project.json` file is where you add more data about your project, sort of like package.json for node.js
+- `LICENSE` file for your code license
+- `README.md` file where you can describe your project
+
+### 🔧 Building the project
+
+Assuming you have successfully initialized a project as seen above, we can now look at how to compile it.
+
+We can simply build it with:
```
Comment (joshring) [inline src/content/docs/Getting Started/projects.mdx]: Let's move the c3c run part so it's first before `c3c build`, so they can see the most important information first

File: src/content/docs/Getting Started/projects.mdx [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+We will now have a binary in build, which we can run:
+
+```bash
+./build/myc3project
+```
+
+It should print `Hello, World!` and return back to the command line prompt. 
+If you are on Windows, you will have `myc3project.exe` instead. Call it in the same way.
+
+### 🏃 Building and running
```
Comment (joshring) [inline src/content/docs/Getting Started/projects.mdx]: See above note about moving this section before the `c3c build`

**Review by joshring — CHANGES_REQUESTED** [review]

**Comment by joshring** [comment]
Thanks it's quite a nifty new feature

It got me thinking about the page names, in addition to above

Shall we change `Hello World in C3` to `Hello World` then it seems more consistent with `Projects`? Or if you have some better titles let's see what you have

I noticed we have Svelte installed but we're not using it, could we nuke that with:
`npm uninstall @astrojs/svelte`

**Comment by tomaskallup** [comment]
I thought about renaming the Hello World to something like `Standalone Compilation`, but I feel like Hello World is so easily recognizable, it's a better name, since it gives you a starting point.

Making it simply `Hello World` is a good idea, there is no need for the suffix.

File: src/content/docs/Getting Started/projects.mdx (hunk 3 above)
Comment (joshring) [reply src/content/docs/Getting Started/projects.mdx:95]: I think the trailing `:::` got a bit lost here, but hopefully you know what I meant

**Review by joshring — APPROVED** [review]

**Comment by tomaskallup** [comment]
Accidentaly clicked on the request review, my page was outdated and I didn't see the approve, sorry for that!

**Comment by lerno** [comment]
Thank you!

**Pull request merged**